llvm/lib/CodeGen/MachineVerifier.cpp

```diff
@@ -1627,18 +1626,18 @@ void MachineVerifier::verifyPreISelGenericInstruction(const MachineInstr *MI) {
       break;
     }
 
+    if (DstTy.getScalarSizeInBits() < 2) {
+      report("result type must be at least 2 bits wide", MI);
+      break;
+    }
+
     if ((DstTy.isVector() != SrcTy.isVector()) ||
         (DstTy.isVector() &&
          DstTy.getElementCount() != SrcTy.getElementCount())) {
       report("Generic vector scmp/ucmp must preserve number of lanes", MI);
       break;
     }
```

Comment:
#120014 removes this check for G_ABDS/G_ABDU. Is it unnecessary or did another PR remove it unnecessarily?

Reply:
Don't mind me. G_ABDS/G_ABDU have the same type for result/operands, G_UCMP/G_SCMP don't and therefore require the check.